griptape/utils/chat.py

```diff
@@ -16,22 +17,39 @@ class Chat:
     prompt_prefix: str = field(default="Q: ", kw_only=True)
     response_prefix: str = field(default="A: ", kw_only=True)
     output_fn: Callable[[str], None] = field(
-        default=Factory(lambda: print), kw_only=True
+        default=lambda x: print(x, end=""), kw_only=True
     )
+    stream: bool = field(default=False, kw_only=True)
+
+    def start(self, initial_turn: str = None) -> None:
+        if initial_turn:
+            self.output_fn(initial_turn + "\n")
+
+        self.structure.stream = self.stream
+        self.structure.prompt_driver.stream = self.stream
+        if self.stream:
+            self.structure = Stream(self.structure)
 
-    def start(self) -> None:
         if self.intro_text:
-            self.output_fn(self.intro_text)
+            self.output_fn(self.intro_text + "\n")
         while True:
             question = input(self.prompt_prefix)
 
             if question.lower() in self.exit_keywords:
-                self.output_fn(self.exiting_text)
-
+                self.output_fn(self.exiting_text + "\n")
                 break
             else:
-                self.output_fn(self.processing_text)
+                self.output_fn(self.processing_text + "\n")
 
-            self.output_fn(
-                f"{self.response_prefix}{self.structure.run(question).output.to_text()}"
-            )
+            if self.stream:
+                self.output_fn(f"{self.response_prefix}")
```

Comment:
Move outside of `self.stream` and remove from line 54.

Reply:
If the prefixes are printed prior to calling run on the agents either mode, you end up with output 
Q: What is the chance martians will invade earth? 
Processing...
A:
and then a delay until it starts getting tokens back. I found this a bit confusing from a UX perspective, not knowing if it had hung. With the two output functions now I think we could leave it as is for non-streaming output and render the response_prefix after the first chunk is returned. I think this is a better UX. What do you think?